### PR TITLE
embed libgraph/idm as a service

### PR DESCRIPTION
**Pull request by rhafer**
## Description
This add a new service "idm" providing and LDAP service (via ldaps) on port 9230.  If not existing it will bootstrap an initial LDAP tree and administrative user as well as a self-signed Certificate and Key (similar to what is done for glauth).

The service is currently not integrated with the other serivce. I.e. changes to point e.g. the graph service and cs3 user and group providers will follow in another PR.

## How Has This Been Tested?
Manual test:

`OCIS_LOG_LEVEL=debug ocis/bin/ocis server`

```
LDAPTLS_REQCERT=allow ldapsearch -x -H ldaps://:9230 -b o=libregraph-idm -Duid=libregraph,o=libregraph-idm -w admin
# extended LDIF
#
# LDAPv3
# base <o=libregraph-idm> with scope subtree
# filter: (objectclass=*)
# requesting: ALL
#

# libregraph-idm
dn: o=libregraph-idm
o: libregraph-idm
objectClass: organization

# users, libregraph-idm
dn: ou=users,o=libregraph-idm
objectClass: organizationalUnit
ou: users

# groups, libregraph-idm
dn: ou=groups,o=libregraph-idm
objectClass: organizationalUnit
ou: groups

# libregraph, libregraph-idm
dn: uid=libregraph,o=libregraph-idm
objectClass: account
objectClass: simpleSecurityObject
uid: libregraph
userPassword:: e0FSR09OMn0kYXJnb24yaWQkdj0xOSRtPTY1NTM2LHQ9MSxwPTIkY2NCV1h1MFF
 yTTI1Q3dLcXVYSkp2QSRoTnVmQkxSUEVQNm40UGU3NUx5ZWVLa0ZBY1ZlNUhNSzFSc3hyQW90TGNZ

# search result
search: 2
result: 0 Success

# numResponses: 5
# numEntries: 4
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: idm/pkg/command/server.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		}
+	}
+	if _, err := os.Stat(servercfg.BoltDBFile); errors.Is(err, os.ErrNotExist) {
+		logger.Debug().Msg("Bootstrapping IDM database")
+		err = bootstrap(logger, cfg, servercfg)
+		logger.Error().Err(err).Msg("failed")
+	}
+
+	svc, err := server.NewServer(&servercfg)
+	svc.Serve(ctx)
```
Comment (kobergj) [inline idm/pkg/command/server.go:75]: Better to call `svc.Serve` after the error check (possible nil pointer dereference)

File: idm/pkg/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+// New initializes the available metrics.
+func New() *Metrics {
+	m := &Metrics{
+		// Counter: prometheus.NewCounterVec(prometheus.CounterOpts{
+		// 	Namespace: Namespace,
+		// 	Subsystem: Subsystem,
+		// 	Name:      "greet_total",
+		// 	Help:      "How many greeting requests processed",
+		// }, []string{}),
```
Comment (kobergj) [inline idm/pkg/metrics/metrics.go:27]: Can we remove the commented code or should we uncomment it? (`Counter` is also commented in the `Metrics` struct)

**Review by kobergj — CHANGES_REQUESTED** [review]

File: idm/pkg/metrics/metrics.go (hunk 2 above)
Comment (rhafer) [reply idm/pkg/metrics/metrics.go:27]: I'll remove it for now. Still need to figure out how we can integrate the metrics that libreraph/idm provides. Into the ocis service.

File: idm/pkg/command/server.go (hunk 1 above)
Comment (rhafer) [reply idm/pkg/command/server.go:75]: Argh, Good catch!

File: idm/pkg/metrics/metrics.go (hunk 2 above)
Comment (rhafer) [reply idm/pkg/metrics/metrics.go]: I've removed the monitoring and tracing pieces for now. They were just stubs anyway

File: idm/pkg/command/server.go (hunk 1 above)
Comment (rhafer) [reply idm/pkg/command/server.go]: Fixed.

**Review by kobergj — APPROVED** [review]

**Pull request merged**